nav2_costmap_2d/test/integration/inflation_tests.cpp

```diff
@@ -146,22 +154,22 @@ void TestNode::validatePointInflation(
         if (cell.x_ > 0) {
           CellData data(costmap->getIndex(cell.x_ - 1, cell.y_),
             cell.x_ - 1, cell.y_, cell.src_x_, cell.src_y_);
-          m[dist].push_back(data);
+          m[distanceFromSource(data)].push_back(data);
```

Comment:
I think this should again be `dist`. The original code has `dist` which is the same as your refactored function. `distFromSource(data)` changes the encoding of information. From your original ticket, you just pointed out that the reference to the iterator needed to be changed.

Reply:
Yeah, I noticed this slightly later so didn't mention it there. 
I does change the encoding of information but shouldn't this new way be the correct way since it is what exactly happens in 
https://github.com/ros-planning/navigation2/blob/6fb43d4a0543897d29d38b3e4a749b6bea6108e1/nav2_costmap_2d/plugins/inflation_layer.cpp#L282-L291

Anyway, I don't think it will make much of a difference so just confirm once finally if you want me to revert this change along with the function. 

The main issue regarding the iterator has been solved so this is not that relevant as you mentioned.

Reply:
I would revert this change to keep things consistent. Especially if you don't think it will effect the outcome. I can follow your logic to remove the iterator, but redoing the logic here would make me have to sit here and sanity check since I'm not familiar with the inflation code (its been around for so, so long).